File: src/main/java/dev/entite/forum/Post.java
package dev.entite.forum;

import com.fasterxml.jackson.annotation.JsonIgnore;
import dev.entite.BaseEntite;
import dev.entite.Utilisateur;

import javax.persistence.*;
/**
 * Classe pour la définition des posts <br/>
 * Se référer à {@link dev.entite.BaseEntite} pour les identifiants générés
 */
@Entity
public class Post extends BaseEntite {

    /**
     * Le contenu du post
     * Lob permet de changer le String "content" de simple "varchar" en "longtext",
     * columnDefinition permet de le changer en "text" <br/>
     */
    @Lob
    @Column(columnDefinition = "TEXT", length = 2048)
    private String content;
    /**
     * Relation many to one avec les utilisateurs <br/>
     * Id dans table post = utilisateur_id<br/>
     * Voir {@link dev.entite.Utilisateur}
     */
    @ManyToOne
    private Utilisateur utilisateur;
    /**
     * Relation many to one avec les topics<br/>
     * Id dans table post = topic_id <br/>
     * Voir {@link dev.entite.forum.Topic}
     */
    @ManyToOne(cascade = {CascadeType.ALL})
    private Topic topic;

    // Constructeur
    public Post() {
    }

    // Getter & Setter
    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Utilisateur getUtilisateur() {
        return utilisateur;
    }

    public void setUtilisateur(Utilisateur utilisateur) {
        this.utilisateur = utilisateur;
    }

    public Topic getTopic() {
        return topic;
    }

    public void setTopic(Topic topic) {
        this.topic = topic;
    }
}

File: src/app/_shared/entities/Post.ts
import {Utilisateur} from "./Utilisateur";
import {Topic} from "./Topic";

export class Post {
  private _id: number
  private _content: string
  private _utilisateur: Utilisateur
  private _topic: Topic
  private _createdDate: string
  private _updatedDate: string


  constructor(id: number, content: string, utilisateur: Utilisateur, topic: Topic, createdDate: string, updatedDate: string) {
    this._id = id;
    this._content = content;
    this._utilisateur = utilisateur;
    this._topic = topic;
    this._createdDate = createdDate;
    this._updatedDate = updatedDate;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get content(): string {
    return this._content;
  }

  set content(value: string) {
    this._content = value;
  }

  get utilisateur(): Utilisateur {
    return this._utilisateur;
  }

  set utilisateur(value: Utilisateur) {
    this._utilisateur = value;
  }

  get topic(): Topic {
    return this._topic;
  }

  set topic(value: Topic) {
    this._topic = value;
  }

  get createdDate(): string {
    return this._createdDate;
  }

  set createdDate(value: string) {
    this._createdDate = value;
  }

  get updatedDate(): string {
    return this._updatedDate;
  }

  set updatedDate(value: string) {
    this._updatedDate = value;
  }
}
